### DISCUSSION TITLE
"Something went wrong - t.roles is not iterable"

### DISCUSSION THREAD

**Discussion by alta-atc** [AntlerVC/firetable#857] [Support / Q&A] [answered]
I started encountering this issue immediately after enabling auditing on a subcollection. No tables in this project will load.

I experimented by changing the audit flag under _rowy_ / settings back to false, and this did not fix the issue. 

![Screenshot 2022-10-06 11 24 23 AM](https://user-images.githubusercontent.com/102269444/194368269-3354d9f8-52e7-489b-99af-7fdf52e415d5.png)

**Comment by alta-atc** [comment, marked as answer]
To the next unfortunate soul to encounter this issue - cast your cares aside. Fix is below

go to rowy / settings in firebase, review the tables and find the ones where the roles array has been deleted. I added the roles array back in (see attachment) and voila, everything is working again.

Potential bug for the rowy team to investigate - when enabling auditing to an existing subcollection, the role can be deleted from the rowy settings document.


![Screenshot 2022-10-06 11 32 28 AM](https://user-images.githubusercontent.com/102269444/194369297-28cec715-81d0-439a-99a6-903a2ef776d1.png)

**Reply by notsidney** [reply]
Thanks @alta-atc, I’ve created an issue for this.

**Reply by notsidney** [reply]
Hi @alta-atc, how were you able to enable auditing for a subcollection? Sub-tables have settings disabled. Unless if you enabled auditing for a collection _group_ instead?

**Reply by alta-atc** [reply]
So I believe I was enabling auditing on a Collection rather than subcollection when this occurred. I have just retraced my steps. Sorry for the confusion.

**Reply by notsidney** [reply]
Are you able to turn on auditing for any other table without it deleting the `roles` array? I haven’t been able to recreate this issue.

Additionally, I’m quite confused as to how you got to this state. How did you create the table in the screenshot?

1. Based on the ID, it looks like you set its Collection to a specific sub-collection in the `vendors` collection.

2. The ID cannot contain `/`. I’ve created a table with `/` in its ID and the UI is not able to load the table at all. Were you able to view that table in the UI at all?

3. The table config in the screenshot does not contain the `collection` field, so it cannot display any data.
